src/server/game/Spells/SpellEffects.cpp

```diff
@@ -5613,17 +5613,16 @@ void Spell::EffectCreateAreaTrigger(SpellEffIndex /*effIndex*/)
     if (effectHandleMode != SPELL_EFFECT_HANDLE_HIT)
         return;
 
+    Unit* target = unitTarget != nullptr ? unitTarget : GetCaster();
```

Comment:
i was talking about SPELL_EFFECT_HANDLE_HIT phase (the right one for this effect)